### PR TITLE
HDDS-8546. Ozone debug chunkinfo command throws GroupMismatchException for closed container

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Fix ozone debug chunkinfo command to work on keys that are present in closed containers, The issue was that the xceiverclient was trying to acquire the original pipeline from the keylocation which gets closed on container/pipeline close. 

1. The fix is to use the separate standalone pipeline instance for read.
```java  ContainerProtos.ContainerDataProto containerData =
      containerOperationClient.readContainer(
          - keyLocation.getContainerID(), keyLocation.getPipeline());
          + keyLocation.getContainerID(), pipeline);
```

2. Also closed client instances by enclosing in try with resources to prevent any leaks.
3. Added TestOzoneDebugShell to test this

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8546

## How was this patch tested?
Added a unit test

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java [hunk 1]
```diff
@@ -71,124 +72,130 @@ public class ChunkKeyHandler extends KeyHandler implements
...
-              responses = null;
-      try {
-        responses = ContainerProtocolCalls.getBlockFromAllNodes(
-            xceiverClient, datanodeBlockID, keyLocation.getToken());
-      } catch (InterruptedException e) {
-        LOG.error("Execution interrupted due to " + e);
-        Thread.currentThread().interrupt();
+    try {
+      containerOperationClient = new ContainerOperationClient(
+          parent.getOzoneConf());
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java:87]: Could be simplified by using `try-with-resources`.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java [hunk 2]
```diff
@@ -71,124 +72,130 @@ public class ChunkKeyHandler extends KeyHandler implements
...
       }
-      responseArrayList.add(responseFromAllNodes);
+      result.add("KeyLocations", responseArrayList);
+      Gson gson2 = new GsonBuilder().setPrettyPrinting().create();
+      String prettyJson = gson2.toJson(result);
+      System.out.println(prettyJson);
+    } finally {
+      xceiverClientManager.close();
+      containerOperationClient.close();
+      ozoneManagerClient.close();
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java:197]: `xceiverClientManager` is closed in `ContainerOperationClient#close`, and I don't think we usually close the `OzoneManagerClient` obtained from `RpcClient`.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneDebugShell.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+import org.apache.hadoop.ozone.client.io.OzoneOutputStream;
+import org.apache.hadoop.ozone.debug.OzoneDebug;
+import org.apache.hadoop.ozone.om.OMConfigKeys;
+import org.apache.hadoop.ozone.om.helpers.OmKeyArgs;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.ozone.test.GenericTestUtils;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneDebugShell.java:44]: Would be nice to use JUnit5 for new tests.

**Review by adoroszlai — COMMENTED** [review]
Thanks @sadanand48 for the patch.  I guess `writeKey()` and `closeContainerForKey()` are duplicated from some other integration tests, would be nice to reuse existing logic.  Otherwise LGTM.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java (hunk 1 above)
Comment (sadanand48) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java]: Done.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java (hunk 2 above)
Comment (sadanand48) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java]: Done.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneDebugShell.java (hunk 3 above)
Comment (sadanand48) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneDebugShell.java]: Done.

**Comment by adoroszlai** [comment]
Thanks @sadanand48 for updating the patch, LGTM.

**Review by dombizita — APPROVED** [review]
thanks for working on this @sadanand48, it looks good to me!

**Comment by adoroszlai** [comment]
Thanks @dombizita for the review.

**Pull request merged**